wp-cli/class-theme-check-cli.php

```diff
@@ -0,0 +1,179 @@
+<?php
+
+if ( defined( 'WP_CLI' ) && WP_CLI ) {
+	$parent_dir = dirname( __DIR__ );
+	require_once $parent_dir . '/checkbase.php';
+	require_once $parent_dir . '/main.php';
+	WP_CLI::add_command( 'theme-check', 'Theme_Check_CLI' );
+}
+
+class Theme_Check_CLI {
+	/**
+	 * Run a theme check on the specified theme or the current theme.
+	 *
+	 * @param array $args       Indexed array of positional arguments.
+	 * @param array $assoc_args Associative array of options.
+	 * @return void
+	 *
+	 * ## OPTIONS
+	 * [<theme>]
+	 * : The slug of the theme to check. If not provided, checks the current theme.
+	 *
+	 * [--format=<format>]
```

Comment:
If you follow this format https://github.com/wp-cli/entity-command/blob/main/src/Post_Command.php#L419 then WP-CLI itself will validate the available options. That way you won't need `in_array()` check in the method.

Reply:
If I removed that snippet:
```
		if ( ! in_array( $format, array( 'cli', 'json' ) ) ) {
			WP_CLI::error( "Invalid format. Accepts 'cli' or 'json'." );
			return;
		}
```
If I use a non-existent format the function runs using the default format instead of raising an error. No matter the comment format implemented in the function. Is that expected?

Reply:
It's not working like that on my end. Removing that snippet and using a random format doesn't throw any error:

![image](https://github.com/user-attachments/assets/a67dab26-28a4-4e67-a9a2-ec3792a9d8d5)

Reply:
It's working fine after implementing your comment formatting suggestion (https://github.com/WordPress/theme-check/pull/458#discussion_r1750604286), thanks!